utils/index.js

```diff
@@ -0,0 +1,37 @@
+const fsPromises = require("fs").promises;
+const StreamZip = require('node-stream-zip');
+
+const readAndParseFile = async (path) => {
+    const data = await fsPromises.readFile(path);
+    return JSON.parse(data);
+};
+
+const readDatasetJson = async (folder) => {
+    const path = `${folder}/dataset.json`;
+    return await readAndParseFile(path);
+};
+
+const readPossibleZippedFile = async (folder, fileName) => {
+    let parsedData;
+    try {
+        const data = await fsPromises.readFile(
+            `${folder}/${fileName}`
+        );
+        parsedData = JSON.parse(data);
```

Comment:
if the file is json but just invalid for some reason, how will we find out?  is the error being totally dropped here?

Reply:
maybe could check first to see if zip exists but not json, then unzip to json.. then read the json?  just a random thought.

Reply:
yeah that's a good point

Reply:
I checked for a "file not found" error, and I also wrapped the zip opening in another try/catch block. I tested it by both messing with a JSON file and deleting a zip file and it errored out appropriately. Let me know what you think